src/nvim/lua/executor.c

```diff
@@ -547,3 +549,37 @@ void ex_luafile(exarg_T *const eap)
     return;
   }
 }
+
+/// Execute ~/.config/nvim/init.lua
+///
+/// @return True if the script was executed, false otherwise.
+bool execute_init_lua(char_u *script_path) {
+  // bail out if the file doesn't exist
+  if (!os_file_is_readable((const char*)script_path)) {
+    return false;
+  }
+
+  if (os_isdir(script_path)) {
```

Comment:
I don't think this is needed: `:luafile directory/` already gives sensible error message from `luaL_loadfile`